File: Chunking/chunk.js
// divide an array into subarrays(chunks). The sub arrays will be the length of the passed in size

function chunk(array, size) {
  const chunks = [];
  let index = 0;

  while(index < array.length) {
    chunks.push(array.slice(index, index + size));
    index += size;
  }
  console.log('chunks', chunks)
}

chunk([1,2,3,4,5,6,7,8], 2);

File: Chunking/Chunk.cs
public List<IEnumerable<T>> Chunk<T>(T[] arr, int size)
{
  var chunks = new List<IEnumerable<T>>();

  for (int i = 0; i < arr.Length / size; i++)
  {
    var chunk = arr.Skip(i * size).Take(size);
    chunks.Add(chunk);
  }

  return chunks;
}